lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2241,6 +2267,26 @@ void TFDeabstraction::formGraphOp(SILTensorOpInfo &opInfo,
       }
       break;
     }
+    case SILTensorOpInfo::OperandClass::ExtractingDTypeList: {
+      // When '$extractingDTypeList' is specified, we are going to recursively
+      // extract all dtypes from the aggregate type and form a list. This
+      // attribute will be canonicalized to a normal attribute.
+      if (constValue.getKind() != SymbolicValue::Metatype)
+        return diagnoseInvalidAttr("requires a metatype");
+      SmallVector<SymbolicValue, 8> metatypes;
+      collectInnermostTensorFlowDTypes(constValue.getMetatypeValue(),
```

Comment:
After you change `collectInnermostTensorFlowDTypes` to return a `bool`, check the condition here. If it failed, diagnose here and say something like "not an aggregate of TensorFlow values".

Reply:
Then, add a diagnostics test in `TensorFlow/diagnostics.swift`.

Reply:
Done